### ISSUE TITLE
Color shortening

### ISSUE THREAD

**Issue by syphen** [dotless/dotless#168]
I know that this implementation shortens colors if they fall into a human readable color (ex: #ffffff returns 'white' instead). This is great for making code smaller, but for implementations with filters (in mixins) the hex code is what is needed to render properly. Can this possible be changed or at least configured as a loadtime setting?

**Comment by rytmis** [comment]
Another thing to consider for the next feature release.

**Comment by rytmis** [comment]
I'm punting on this one for now, because I can't convince myself that I understand the implications of changing the behavior yet, and I really want to get 1.5.0 out sooner rather than later.

**Comment by nimeshdhruve** [comment]
I am in favor of not shortening the color. I am using WebCompiler which internally uses dotless. [Here is the original issue](https://github.com/madskristensen/WebCompiler/issues/53) that I reported.

**Comment by rytmis** [comment]
I'm not arguing against changing the logic, but I'm curious: why is this a problem for you? The colors are only changed when the keyword matches the color RGB value, and dotless can operate on the keyworded colors just as well as RGB values.

**Comment by nimeshdhruve** [comment]
This is not a problem (but little annoyance) for me. It would be great to let the developer decide what they want rather then forcing them :) 

Recently I started using WebCompiler for processing my LESS files and to verify that it is working fine I wanted to compare result with previously processed file and new file, now as you can imagine previous file was having hex values and new one was color name so there were way too many differences and honestly it was pain in neck to verify these two versions :)

**Comment by rytmis** [comment]
Ah, right. I can see how that might be a problem. Well, since this still seems to be a pain point, and even Luke seemed to be leaning towards this, I'll take this up on my list of things to do for the next feature release.

**Comment by lukeapage** [comment]
less. js currently remembers the input form and outputs the same. The primary reason is string interpolation, keeping strimgs consistent.

**Comment by rytmis** [comment]
Yeah, I figured that would be the case -- it makes sense to do it like that. :)

**Comment by yln** [comment]
Are we still interested in aligning our output with less.js? If so, I can take a look at this.

less.js seems to never change/shorten colors and even outputs the long hex form for computed values, i.e., `darken(blue, 10%)` becomes `#0000cc` (not `#00c`).

**Comment by rytmis** [comment]
Absolutely, go ahead. Let's remember the input form. :)
